tests/test_regrid/test_structured.py

```diff
@@ -0,0 +1,370 @@
+import numpy as np
+import pytest
+import xarray as xr
+
+from xugrid.regrid.structured import StructuredGrid1d, StructuredGrid2d
+
+# Testgrids
+# --------
+# grid a(x):               |______50_____|_____100_____|_____150_____|               -> source
+# grid b(x):        |______25_____|______75_____|_____125_____|_____175_____|        -> target
+# --------
+# grid aa(x):                       |______80_____|_____130_____|_____180_____|      -> source
+# grid bb(x):            |______40_____|______90_____|_____140_____|____190_____|    -> target
+# --------
+# grid a(x):               |______50_____|_____100_____|_____150_____|               -> source
+# grid bbb(x):              |__30__|__55__|__80_|__105__|                            -> target
```

Comment:
it seems to me that all the test cases are regular grids (same cell size). That can obfuscate some errors in the code, please make one of these (or a new one) irregular